packages/templates/blog/template/src/bootstrap.js

```diff
@@ -50,6 +50,17 @@ async function setPublicPermissions(newPermissions) {
   await Promise.all(allPermissionsToCreate);
 }
 
+async function setInitFrontendAPIToken(attributes) {
+  const apiTokenService = strapi.service(`admin::api-token`);
+  const apiToken = await apiTokenService.create(attributes);
+  const accessKey = apiToken.accessKey;
+  const credentialsLocation = process.env.FRONTEND_ENV_LOCATION || "../frontend/.env.development"
```

Comment:
Many people use templates without starters, so I think this is too risky. Even if `../frontend/.env.development` exists, we have no guarantee that it's the matching starter for a template, so we could be writing things in an unrelated folder, which we don't want.

I think instead we should only run `setInitFrontendAPIToken` if we know it's a starter using an env variable. But instead of FRONTEND_ENV_LOCATION or DEMO_IS_STARTER, I'd recommend using `STARTER_FRONTEND_PATH`: it tells us both that it's a starter, and where to look for the frontend.

>If I had a better idea of where/how the starting .env file was generated from

Indeed we'd have to programmatically generate this env variable. This would be done in the [create-strapi-starter package](https://github.com/strapi/strapi/tree/main/packages/cli/create-strapi-starter)

Reply:
@remidej I just checked back in here and was about to get back to trying to implement a gatsby app with the strapi backend. I was following along on this issue [here](https://github.com/strapi/starters-and-templates/issues/38) and saw the PR. 

I am going to attempt to get it working trying @chadwcarlson 's work-around. I'm not super familiar with the codebase, but would be happy to help if it could get the gatsby integration working and still maintain security. 👍🏽